### PR TITLE
M2

### PR DESCRIPTION
**Pull request by 0xPxt**
Parser working, fuzzer not showing errors.
Opened PR anyway to speed up the process.

### CODE REVIEW

File: app/src/items.c [hunk 1]
```diff
@@ -0,0 +1,522 @@
...
+    item_array.numOfUnknownCapabitilies = 1;
+
+    for (uint8_t i = 0; i < sizeof(item_array.items) / sizeof(item_array.items[0]); i++) {
+        item_array.items[i].can_display = bool_true;
+    }
+}
+
+item_array_t *items_getItemArray() { return &item_array; }
+
+void items_storeItems() {
```
Comment (neithanmo) [inline app/src/items.c]: items_storeItems should return an error, something like:
```c
items_error_t items_storeItems() {
     CHECK_ERROR(items_storeSigningTransaction());
     CHECK_ERROR(items_storeNetwork());
```
and so on?

File: app/src/items_defs.h [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+#define ITEMS_TO_PARSER_ERROR(__CALL)                          \
+    {                                                          \
+        items_error_t __err = __CALL;                          \
+        CHECK_APP_CANARY()                                     \
+        if (__err != items_ok) return parser_unexpected_error; \
+    }
+
+typedef struct {
+    char key[23];
```
Comment (neithanmo) [inline app/src/items_defs.h]: Let move 23 to a constant definition, with a comment on why that value??

File: app/src/json/json_parser.c [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+    if (array_token_index > json->numberOfTokens) {
+        return parser_no_data;
+    }
+
+    jsmntok_t array_token = json->tokens[array_token_index];
+    *token_index = array_token_index;
+
+    uint16_t element_count = 0;
+    uint16_t prev_element_end = array_token.start;
+    while (true) {
```
Comment (neithanmo) [inline app/src/json/json_parser.c]: why not using the token_index condition directly here? this works anyways just saying :-)

Comment (neithanmo) [reply app/src/json/json_parser.c]: while(*token_index < json->numberOfTokens)

**Review by neithanmo — APPROVED** [review]

File: app/src/items.c (hunk 1 above)
Comment (0xPxt) [reply app/src/items.c]: Wouldn't that make the parsing stop if anything goes wrong? According to the test vectors, when there is an invalid value we should either not display it, show it as unknown capability or raise a warning, depending on the value. What do you think?

**Pull request merged**